NomadUI/Core/NUIScrollbar.cpp

```
@@ -136,31 +162,101 @@ bool NUIScrollbar::onMouseEvent(const NUIMouseEvent& event)
+            if (availableTrack > 0.5 && availableValue > 0.0) {
+                double valueDelta = (deltaPixels / availableTrack) * availableValue;
+                double newValue = dragStartValue_ + valueDelta;
+                scrollTo(newValue);
+            }
+        } else if (pressedPart_ == Part::ThumbStartEdge) {
+            // Dragging left/top edge: changes start and size
+            // Use relative movement (step delta) to handle dynamic range changes during drag
+            bool isTimelineResize = (style_ == Style::Timeline);
+            
+            double stepDeltaPixels;
+            if (orientation_ == Orientation::Vertical) stepDeltaPixels = event.position.y - lastMousePosition_.y;
```

Clang (14.0.6)

[note] +3, including nesting penalty of 2, nesting level increased to 3

(clang)